File: gemini_prompt.py
import json
import sys
from pathlib import Path

from google import genai


DEFAULT_PROMPT = "Explain how AI works"
CONFIG_PATH = Path(__file__).with_name("config.json")


def load_config():
    with CONFIG_PATH.open(encoding="utf-8") as config_file:
        return json.load(config_file)


def main():
    config = load_config()
    api_key = config.get("apiKey")
    if not api_key:
        print("Missing apiKey in config.json.", file=sys.stderr)
        sys.exit(1)

    model = config["model"]
    prompt = " ".join(sys.argv[1:]) or DEFAULT_PROMPT

    client = genai.Client(api_key=api_key)
    response = client.models.generate_content(
        model=model,
        contents=prompt,
    )

    print(response.text)


if __name__ == "__main__":
    main()

File: gemini_prompt.js
import { GoogleGenAI } from "@google/genai";
import fs from "node:fs";

const DEFAULT_PROMPT = "Explain how AI works in a few words";
const CONFIG_PATH = new URL("./config.json", import.meta.url);

function loadConfig() {
  return JSON.parse(fs.readFileSync(CONFIG_PATH, "utf8"));
}

async function main() {
  const config = loadConfig();
  const apiKey = config.apiKey;

  if (!apiKey) {
    console.error("Missing apiKey in config.json.");
    process.exit(1);
  }

  const model = config.model;
  const prompt = process.argv.slice(2).join(" ") || DEFAULT_PROMPT;

  const ai = new GoogleGenAI({ apiKey });
  const response = await ai.models.generateContent({
    model,
    contents: prompt,
  });

  console.log(response.text);
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
